code/modules/mob/observer/ghost/ghost.dm

```diff
@@ -615,3 +763,58 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 	if((!target) || (!ghost)) return
 	. = "<a href='byond://?src=\ref[ghost];track=\ref[target]'>follow</a>"
 	. += target.extra_ghost_link(ghost)
+
+
+/mob/observer/proc/initialise_postkey()
+	//This function should be run after a ghost has been created and had a ckey assigned
+	//Death times are initialised if they were unset
+	//get/set death_time functions are in mob_helpers.dm
+	//These initialised times are designed to allow a player to immediately be a mouse or drone if they joined as observer from lobby
+	if (!get_death_time(ANIMAL))
+		set_death_time(ANIMAL, world.time - ANIMAL_SPAWN_DELAY)//allow instant mouse spawning
+	if (!get_death_time(MINISYNTH))
+		set_death_time(MINISYNTH, world.time - DRONE_SPAWN_DELAY) //allow instant drone spawning
+	if (!get_death_time(CREW))
+		set_death_time(CREW, world.time)
+
+
+/mob/verb/abandon_mob()
```

Comment:
SUGG: make it `/mob/overver/verb` inster `/mob/verb`

Reply:
fwiw i didn't create that, just moved it

it already has plenty of checks and it appears in the ooc menu, there's not much need to restrict who can use it

Reply:
AFAIR there was runtime errors with `/mob/new_player` in that verb.
And only observers actually need that verb.

Reply:
It's still useful to have it visible so people can learn that it exists. Ill put in a check to exclude mob/new_player to fix any runtimes